opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -371,6 +375,72 @@ def detect(self) -> "Resource":
         return Resource(resource_info)
 
 
+class OsResourceDetector(ResourceDetector):
+    """Detect os resources based on `Operating System conventions <https://opentelemetry.io/docs/specs/semconv/resource/os/`_."""
+
+    def detect(self) -> "Resource":
+        """Returns a resource with with `os.type` and `os.version`. Example of
+        return values for `platform` calls in different systems:
+
+        Linux:
+        >>> platform.system()
+        'Linux'
+        >>> platform.release()
+        '6.5.0-35-generic'
+        >>> platform.version()
+        '#35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2'
+
+        MacOS:
+        >>> platform.system()
+        'Darwin'
+        >>> platform.release()
+        '23.0.0'
+        >>> platform.version()
+        'Darwin Kernel Version 23.0.0: Fri Sep 15 14:42:57 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T8112'
+
+        Windows:
+        >>> platform.system()
+        'Windows'
+        >>> platform.release()
+        '2022Server'
+        >>> platform.version()
+        '10.0.20348'
+
+        FreeBSD:
+        >>> platform.system()
+        'FreeBSD'
+        >>> platform.release()
+        '14.1-RELEASE'
+        >>> platform.version()
+        'FreeBSD 14.1-RELEASE releng/14.1-n267679-10e31f0946d8 GENERIC'
+
+        Solaris:
+        >>> platform.system()
+        'SunOS'
+        >>> platform.release()
+        '5.11'
+        >>> platform.version()
+        '11.4.0.15.0'
+        """
+
+        os_type = platform.system().lower()
+        os_version = platform.release()
+
+        # See docstring
+        if os_type == "windows":
+            os_version = platform.version()
+        # Align SunOS with conventions
+        if os_type == "sunos":
```

Comment:
What do you think on passing `aliased=True` to platform.platform (https://github.com/python/cpython/blob/main/Lib/platform.py#L1337) to get solaris version handling too per https://github.com/python/cpython/blob/main/Lib/platform.py#L611 ?

Reply:
Ah we can't because macos is reported as such and not darwin.